flink-core/src/test/java/org/apache/flink/api/common/typeutils/base/array/PrimitiveArraySerializerSnapshotMigrationTest.java

```diff
@@ -18,77 +18,85 @@
 
 package org.apache.flink.api.common.typeutils.base.array;
 
-import org.apache.flink.api.common.typeutils.TypeSerializerSnapshotMigrationTestBase;
+import org.apache.flink.api.common.typeutils.TypeSerializerUpgradeTestBase;
 import org.apache.flink.testutils.migration.MigrationVersion;
 
 import org.junit.runner.RunWith;
 import org.junit.runners.Parameterized;
 
+import java.util.ArrayList;
 import java.util.Collection;
 
 /**
  * Migration tests for primitive array type serializers' snapshots.
  */
 @RunWith(Parameterized.class)
-public class PrimitiveArraySerializerSnapshotMigrationTest extends TypeSerializerSnapshotMigrationTestBase<Object> {
-
-	public PrimitiveArraySerializerSnapshotMigrationTest(TestSpecification<Object> testSpecification) {
+public class PrimitiveArraySerializerSnapshotMigrationTest extends TypeSerializerUpgradeTestBase<Object, Object> {
```

Comment:
Could you please also rename the serializers to conform to the `FooSerializerUpgradeTest` schema? That's what I have used so far for the other serializers.